src/app/App.tsx

```diff
@@ -1,7 +1,30 @@
+import { useThemeStore } from '@/store/theme';
+import { useEffect } from 'react';
+
 import AppProvider from './AppProvider';
 import AppRouter from './routes/AppRouter';
 
 function App() {
+  const theme = useThemeStore((state) => state.theme);
+
+  useEffect(() => {
+    const root = window.document.documentElement;
+
+    root.classList.remove('light', 'dark');
+
+    if (theme === 'system') {
+      const systemTheme = window.matchMedia('(prefers-color-scheme: dark)')
+        .matches
+        ? 'dark'
+        : 'light';
+
+      root.classList.add(systemTheme);
+      return;
+    }
+
+    root.classList.add(theme);
+  }, [theme]);
+
```

Comment:
To byloby fajnie wynieść do jakeigoś osobnego hooka, który będzie tylko to zawierał i zwróci <></> lub nulla, bo szkoda zaśmiecać App taką logiką.

Tu masz np. bardzo fajny sposób na dark mode https://jskim1991.medium.com/react-how-to-implement-color-themes-using-scss-f8a774ee0f20